ui/src/app/utils/research-purpose.tsx

```diff
@@ -43,9 +44,9 @@ export function getSelectedResearchPurposeItems(researchPurpose: ResearchPurpose
 }
 
 export function getSelectedPopulations(researchPurpose: ResearchPurpose) {
-  const categories = SpecificPopulationItems.filter(specificPopulationItem => specificPopulationItem
-    .subCategory.filter(item => researchPurpose.populationDetails.includes(item.shortName)).length > 0);
-  categories.forEach(category => category.subCategory = category.subCategory
+  const categories = fp.cloneDeep(SpecificPopulationItems.filter(specificPopulationItem => specificPopulationItem
```

Comment:
Why do we need to deep clone this? Now that its a filter instead of a forEach?

Reply:
the logic within forEach was updating the actually SpecificPopulationItems's subCateogries. It was removing all the subCateogories that were not selected.

Reply:
I understand that bit, I was just wondering about the deep clone that its doing. Do we need to generate the copy?

Reply:
had an offline discussion with @blrubenstein , we are using deep clone approach here because we want to Filter subCategories from Categories object, i.e we want to filter out nested Object hence making a new copy and manipulating it makes more sene